packages/consoledot-api/src/mutations/useUpdateTopicMutation.ts

```diff
@@ -0,0 +1,62 @@
+import { isServiceApiError } from "@rhoas/kafka-management-sdk";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { kafkaQueries } from "../queryKeys";
+import { useApi } from "../useApi";
+import type { TopicSettings } from "@rhoas/kafka-instance-sdk";
+import type { Topic } from "ui-models/src/models/topic";
+import { convertToKeyValuePairs } from "consoledot-api/src/transformers/topicTransformer";
+
+export function useUpdateTopicMutation() {
+  const { topics } = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    async function UpdateTopicMutation({
+      adminUrl,
+      topic,
+      onSuccess,
+      onError,
+    }: {
+      instanceId: string;
+      adminUrl: string;
+      topic: Topic;
+      onSuccess: () => void;
+      onError: (code: string, message: string) => void;
+    }) {
+      const api = topics(adminUrl);
+
+      const convertToNewTopicInput = (topic: Topic) => {
+        const { partitions, ...config } = topic;
+
+        const configEntries = convertToKeyValuePairs(config);
+        const topicSettings: TopicSettings = {
+          numPartitions: partitions.length,
+          config: configEntries,
+        };
+        return topicSettings;
+      };
```

Comment:
This is _almost_ duplicated, it's already defined in `useCreateTopicMutation` with the difference of the `name` property. Move it to the transformers and just import it in both hooks and handle the `name` where it's needed.